examples/AcousticWave2D_codegen.py

```diff
@@ -95,21 +95,43 @@ def _init_taylor(self, nt):
         # domain to the border
 
         # Forward wave equation
+
+        def damp_boundary(damp):
+            h = self.h
+            dampcoeff = 1.5 * np.log(1.0 / 0.001) / (40 * h)
+            nx, ny = self.model.get_shape()
+            nbpml = self.nbpml
+            for i in range(nbpml):
+                pos = np.abs((nbpml-i)/nbpml)
+                val = dampcoeff * (pos - np.sin(2*np.pi*pos)/(2*np.pi))
+                damp[i, :] += val
+                damp[-i, :] += val
+                damp[:, i] += val
+                damp[:, -i] += val
+
         wave_equation = m * dtt - (dxx + dyy) + e * dt
         stencil = solve(wave_equation, p(x, y, t+s))[0]
         self.nt = nt
-
+        m_sub = DenseData("m", self.model.vp.shape, self.dtype)
+        m_sub.initializer = lambda ref: np.copyto(ref, self.model.vp**(-2))
```

Comment:
In that case, simply replace it with what you would otherwise pass into the `initialisze`, either `self.data = self.model.vp**-2` or `self.data = self.model.vp.copy()`.